core/src/main/java/org/apache/calcite/runtime/SpatialTypeFunctions.java

```diff
@@ -1318,6 +1322,195 @@ public static Geometry ST_Translate(Geometry geom, BigDecimal x, BigDecimal y) {
     return transformation.transform(geom);
   }
 
+  // Geometry measurement functions
+
+  /**
+   * Returns the area of the {@code geom}.
+   */
+  public static @Nullable Double ST_Area(Geometry geom) {
+    return geom.getArea();
```

Comment:
Just curious: why `ST_Area` and  `ST_Length` don't have a null-check and the other methods do?

Reply:
This is a good point, I fixed these two methods. I started adding test cases for null values after realizing that they were supported by postgis. I will open an issue to ensure that we are consistent when it come to null values in the other methods.

Reply:
After reading a the javadoc for [SqlFunctions](https://javadoc.io/doc/org.apache.calcite/calcite-core/1.26.0/org/apache/calcite/runtime/SqlFunctions.html), it looks like checks for nulls are performed by Calcite before calling the functions. From what I understand, the default behavior is as follow: the functions always return null if one or more of its arguments are null. As this behavior looks similar to what postgis does, I removed all my null checks. Can you confirm that this is the right approach?

Reply:
Thanks a lot for the clarification and the pointers.